.github/workflows/test.yml

```
@@ -0,0 +1,90 @@
+name: build-and-test
+on: [pull_request, push]
+jobs:
+  build-test-repo:
+    strategy:
+      fail-fast: false
+      matrix:
+        cfg:
+          - runs-on: ubuntu-latest
+            cmargs: >
+              -DLAPACK_LIBRARIES=${CONDA_PREFIX}/lib/libmkl_rt.so
+          - runs-on: macos-latest
+    name: "${{ matrix.cfg.runs-on }}"
+    runs-on: ${{ matrix.cfg.runs-on }}
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+    - name: Checkout Ambit
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Write a Conda Env File
+      run: |
+        cat > export.yaml <<EOF
+        name: test
+        channels:
+          - conda-forge
+        dependencies:
+          - cmake >=3.16
+          - ninja
+          - cxx-compiler
+          - pybind11 >=2.6.2
+          - hdf5 >=1.10.6
+          - mkl >=2021.4.0
+          - pytest
+          - numpy
+        EOF
+        if [[ "${{ runner.os }}" == "Windows" ]]; then
+          sed -i "s/- cxx/#- cxx/g" export.yaml
+        fi
+        cat export.yaml
+    - name: Create Conda Environment
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        mamba-version: "*"
+        activate-environment: test
+        channels: conda-forge
+        environment-file: export.yaml
+        show-channel-urls: true
+    - name: Environment Information for Debug
+      run: |
+        conda info
+        conda list
+    - name: Extract, Build, Install Ambit
+      run: |
+        unset LDFLAGS
+        cmake \
+          -S. \
+          -Bbuild \
+          -G"Ninja" \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DCMAKE_INSTALL_PREFIX="${{github.workspace}}/installed" \
+          -DCMAKE_CXX_COMPILER=${CXX} \
+          -DCMAKE_PREFIX_PATH="${CONDA_PREFIX}" \
```

```suggestion
          -S . \
          -B build \
          -G "Ninja" \
          -D CMAKE_BUILD_TYPE=Release \
          -D CMAKE_INSTALL_PREFIX="${{github.workspace}}/installed" \
          -D CMAKE_CXX_COMPILER=${CXX} \
          -D CMAKE_PREFIX_PATH="${CONDA_PREFIX}" \
```
this doesn't matter, of course, but it's handy for full-word searching to have the cmake var names in the same form as in the CMakeLists.txt file.